### DISCUSSION TITLE
Problem with iPhone VPN connection

### DISCUSSION THREAD

**Discussion by chistyi** [strongswan/strongswan#2363] [Q&A] [answered]
Hello!

I have a problem connecting to my VPN server from an iPhone.
The connection does not want to be established.
Output on the iPhone: first there is a connection, then a disconnection and that's it
I can't find anything in the log that would indicate a problem.
I am attaching the configuration, log, mobileconfig and swanсtl -l/L output.

[swanctl.conf](https://github.com/user-attachments/files/16340002/swanctl.conf.txt)
[swanctl -lL](https://github.com/user-attachments/files/16340003/swanctl.-lL.txt)
[charon_debug.log](https://github.com/user-attachments/files/16339990/charon_debug.log)
[test-user.mobileconfig](https://github.com/user-attachments/files/16339997/test-user.mobileconfig.txt)

**Comment by tobiasbrunner** [comment]
I don't see what you describe in the log. It ends with the connection fully established (IKE and Child SA).

**Reply by chistyi** [reply]
Yes, in the log everything looks like a successfully created connection.
But the connection is not established on my phone.

**Reply by tobiasbrunner** [reply]
I see. The client does not send a certificate request, so no server certificate is sent. The client will need that to verify the server's authentication. Try setting `send_cert=always` to force sending the server certificate.

**Comment by chistyi** [comment]
With the `send_cert=always` parameter I have some changes.
Line:
`configuration payload negotiation failed, no CHILD_SA built`
And some other lines.

Full log:
[charon_debug.log](https://github.com/user-attachments/files/16353547/charon_debug.log)

**Reply by tobiasbrunner** [reply]
You must have messed up in your config when you changed that. It sounds like you removed the `pools = rw_pool` setting. Please post your config.

**Comment by chistyi** [comment]
Yeah, I messed up the config (use **poolls** not pools), i fixed it.
And again according to the logs everything looks like the connection is established. 
But the phone is not connected.

[charon_debug.log](https://github.com/user-attachments/files/16359437/charon_debug.log)
[swanctl.conf](https://github.com/user-attachments/files/16359448/swanctl.conf.txt)

**Reply by tobiasbrunner** [reply]
Maybe something with your certificate(s) is wrong. Please post the output of `swanctl --list-certs`.

**Reply by chistyi** [reply]
Maybe the phone doesn`t want to work with rsa2048 certs?
I will try to regenerate them a little later.

[swanctl --list-certs.txt](https://github.com/user-attachments/files/16360077/swanctl.--list-certs.txt)

**Reply by tobiasbrunner** [reply]
I don't think RSA is an issue. But since you use a custom CA, do you have that installed on your client? Also, using the same subject DN for the CA and the end-entity certificate(s) is not a good idea, makes debugging tricky and some clients might even have problems with it.

**Reply by chistyi** [reply]
This problem is solved!
Thanks!
It was a problem with certificates. I need to use server IP (or maybe FQND) in CA certificate CN subject: `C=RU, O=Pudge, CN=93.183.127.172`. 
And I added serverAuth and clientAuth flags to client and server certs (I don't know if it helped, but nothing worked until I added ip to CA certificate CN).

[swanctl --list-certs.txt](https://github.com/user-attachments/files/16366784/swanctl.--list-certs.txt)

**Reply by tobiasbrunner** [reply, marked as answer]
No, in the CA certificate that's definitely not necessary as that certificate is not associated with a specific server. You need the server's identity (IP or FQDN) as subjectAltName in the server certificate (you already have the IP there), but the subject DN there doesn't matter that much either as there could be multiple different subjectAltNames. The flags are actually more likely the reason it works now.

**Reply by chistyi** [reply]
You were right, i had:
`<key>ServerCertificateIssuerCommonName</key>`
`<string>x.x.x.x</string>`
in my mobileconfig file and it ruined my connections with different CA CN.

**Comment by chistyi** [comment]
But i have one more question.
I have a problem with multiple connections from one NAT, but not always.
When i use home internet provider for multiple connections my NAT IP reserve same ports (4500 port) for different connections (full in --list-sas (wifi).txt file):
`remote 'user-2' @ y.y.y.y[4500] [10.10.10.2]
remote 'user-1' @ y.y.y.y[4500] [10.10.10.1]`
[--list-sas (wifi).txt](https://github.com/user-attachments/files/16366919/--list-sas.wifi.txt)
This behavior breaks one of the connections.

But when i use mobile internet provider my NAT IP reserve random port (not 4500):
`remote 'user-1' @ 188.170.72.139[18235] [10.10.10.2]`
[--list-sas (mobile).txt](https://github.com/user-attachments/files/16367007/--list-sas.mobile.txt)

**Reply by tobiasbrunner** [reply]
Yeah, that won't work. If your NAT router does that, it behaves incorrectly and there is nothing you can do on the server side. Maybe check if the router has something like an "IPsec passthrough" feature that you can disable (it might treat port 4500 in a special way and forward all traffic to the last client that used the port).

**Reply by chistyi** [reply]
Ok, thank you very much!
